application/helpers/frontend_helper.php

```diff
@@ -1865,6 +1865,10 @@ function doAssessment($surveyid, $returndataonly=false)
             $groups=array();
             foreach($fieldmap as $field)
             {
+                // Init Assessment Value
+                $assessmentValue = NULL;
```

Comment:
It did searched db for all subquestions.
This covers all question types, which gives lot of felxibility.